client/router_update.go

```diff
@@ -455,6 +461,35 @@ func (cli *VanClient) RouterUpdateVersionInNamespace(ctx context.Context, hup bo
 		kube.AppendSharedVolume(&router.Spec.Template.Spec.Volumes, []*[]corev1.VolumeMount{&router.Spec.Template.Spec.Containers[0].VolumeMounts, &router.Spec.Template.Spec.Containers[1].VolumeMounts}, "skupper-router-certs", "/etc/skupper-router-certs")
 		updateRouter = true
 	}
+	if addClaimsSupport {
+		err = kube.UpdateRole(namespace, types.TransportRoleName, types.TransportPolicyRule, cli.KubeClient)
+		if err != nil {
+			return false, err
+		}
+		if !config.IsEdge() {
+			if usingRoutes {
+				err = cli.createClaimsRedemptionRoute(ctx, namespace)
+				if err != nil {
+					return false, err
+				}
+			}
+		}
+	}
+	if moveClaims {
+		if err := cli.moveClaimsToRouterService(ctx, namespace); err != nil {
+			return false, err
+		}
+		latest, err := kube.WaitDeploymentReady(types.TransportDeploymentName, namespace, cli.KubeClient, time.Minute, time.Second)
+		if err != nil {
+			return false, err
+		}
+		if latest.ObjectMeta.ResourceVersion != router.ObjectMeta.ResourceVersion {
+			latest.Spec = router.Spec
+			router = latest
+		}
+		kube.AppendSecretVolumeWithVolumeName(&router.Spec.Template.Spec.Volumes, &router.Spec.Template.Spec.Containers[1].VolumeMounts, types.SiteServerSecret, "claims-cert", "/etc/skupper-internal/")
```

Comment:
There is one problem here. If, let's say we are updating from 1.4.2 to 1.4.3, this will work fine.
If we need to update from 1.4.3 to 1.4.4, the secret will be appended twice and will cause the
router update to fail.

Reply:
This code will only run when updating to 1.5.x, and only when the original version is less than 1.5.0. So e.g. if you were at 1.4.2 and used the 1.5.0 client t update, the site would then be 1.5.0. If you then updated to 1.5.1 then the moveClaims flag would be false so you wouldn't append again.

Or am I missing the sequence that would cause the problem? (It would be fairly easy to fix anyway, I just don't yet see how that line would run twice).

Reply:
Alright. My bad. I thought that, in spite of it being clear that applies only to sites using less than 1.5.0,
it could be part of an eventual upcoming patch release. So if this is just going to land as part of 1.5.0,
then it is working just fine.
Sorry for the noises and confusion.